pkg/providers/vsphere/reconciler/reconciler.go

```diff
@@ -113,181 +96,57 @@ func (r *Reconciler) Reconcile(ctx context.Context, log logr.Logger, cluster *an
-		if err != nil {
-			return controller.Result{}, err
-		}
-
-		log.Info("About to apply CNI")
-
-		helm := executables.NewHelm(executables.NewExecutable("helm"), executables.WithInsecure())
-		cilium := cilium.NewCilium(nil, helm)
-
-		if err != nil {
-			return controller.Result{}, err
-		}
-		ciliumSpec, err := cilium.GenerateManifest(ctx, specWithBundles, []string{constants.CapvSystemNamespace})
-		if err != nil {
-			return controller.Result{}, err
-		}
-		if err := serverside.ReconcileYaml(ctx, remoteClient, ciliumSpec); err != nil {
-			return controller.Result{}, err
-		}
-		conditions.MarkTrue(cluster, cniSpecAppliedCondition)
-	}
+func (r *Reconciler) ReconcileCNI(ctx context.Context, log logr.Logger, clusterSpec *c.Spec) (controller.Result, error) {
+	log = log.WithValues("phase", "reconcileCNI")
+	// TODO: implement CNI reconciliation phase
 	return controller.Result{}, nil
 }
 
-func (r *Reconciler) getCAPICluster(ctx context.Context, cluster *anywherev1.Cluster, log logr.Logger) (*clusterv1.Cluster, controller.Result, error) {
-	capiCluster := &clusterv1.Cluster{}
-	capiClusterName := types.NamespacedName{Namespace: constants.EksaSystemNamespace, Name: cluster.Name}
-	log.Info("Searching for CAPI cluster", "name", cluster.Name)
-	if err := r.client.Get(ctx, capiClusterName, capiCluster); err != nil {
-		return nil, controller.Result{Result: &ctrl.Result{
-			Requeue:      true,
-			RequeueAfter: defaultRequeueTime,
-		}}, err
-	}
-	return capiCluster, controller.Result{}, nil
+func (r *Reconciler) ReconcileWorkers(ctx context.Context, log logr.Logger, clusterSpec *c.Spec) (controller.Result, error) {
+	log = log.WithValues("phase", "reconcileWorkers")
+	// TODO: implement workers reconciliation phase
+	return controller.Result{}, nil
 }
 
-func (r *Reconciler) reconcileWorkerNodeSpec(
-	ctx context.Context, cluster *anywherev1.Cluster, templateBuilder providers.TemplateBuilder,
-	specWithBundles *c.Spec, workloadTemplateNames, kubeadmconfigTemplateNames map[string]string,
-) (controller.Result, error) {
-	if !conditions.IsTrue(cluster, workerNodeSpecPlaneAppliedCondition) {
-		workersSpec, err := templateBuilder.GenerateCAPISpecWorkers(specWithBundles, workloadTemplateNames, kubeadmconfigTemplateNames)
-		if err != nil {
-			return controller.Result{}, err
-		}
+func (r *Reconciler) validateMachineConfigs(ctx context.Context, cluster *anywherev1.Cluster, clusterSpec *c.Spec, datacenterConfig *anywherev1.VSphereDatacenterConfig) error {
+	machineConfigMap := map[string]*anywherev1.VSphereMachineConfig{}
 
-		if err := serverside.ReconcileYaml(ctx, r.client, workersSpec); err != nil {
-			return controller.Result{}, err
+	for _, ref := range cluster.MachineConfigRefs() {
+		machineConfig := &anywherev1.VSphereMachineConfig{}
+		machineConfigName := types.NamespacedName{Namespace: cluster.Namespace, Name: ref.Name}
+		if err := r.client.Get(ctx, machineConfigName, machineConfig); err != nil {
```

Comment:
The machine configs are already in the cluster spec, we can just get them from there